examples/30_extended/custom_flow_tutorial.py

```diff
@@ -0,0 +1,183 @@
+and also show how to link runs to the custom flow.
+"""
+
+####################################################################################################
+
+# License: BSD 3-Clause
+# .. warning:: This example uploads data. For that reason, this example
+#   connects to the test server at test.openml.org. This prevents the main
+#   server from crowding with example datasets, tasks, runs, and so on.
+from collections import OrderedDict
+import numpy as np
+
+import openml
+from openml import OpenMLClassificationTask
+from openml.runs.functions import format_prediction
+
+openml.config.start_using_configuration_for_example()
+
+####################################################################################################
+# 1. Defining the flow
+# ====================
+# The first step is to define all the hyperparameters of your flow.
+# Check ... for the descriptions of each variable.
+# Note that `external version` and `name` together should uniquely identify a flow.
+#
+# The AutoML Benchmark runs AutoML systems across a range of tasks.
+# We can not use the flows of the AutoML systems directly, as the benchmark adds performs
+# preprocessing as required.
+#
+# We will break down the flow parameters into several groups, for the tutorial.
+# First we will define the name and version information.
+# Make sure to leave enough information so others can determine exactly which
+# version of the package/script is used. Use tags so users can find your flow easily.
+
+general = dict(
+    name="automlbenchmark_autosklearn",
+    description=(
+        "Auto-sklearn as set up by the AutoML Benchmark"
+        "Source: https://github.com/openml/automlbenchmark/releases/tag/v0.9"
+    ),
+    external_version="amlb==0.9",
+    language="English",
+    tags=["amlb", "benchmark", "study_218"],
+    dependencies="amlb==0.9",
+)
+
+####################################################################################################
+# Next we define the flow hyperparameters. We define their name and default value in `parameters`,
+# and provide meta-data for each parameter through `parameters_meta_info`.
+# Note that the use of ordered dicts is required.
+
+flow_hyperparameters = dict(
+    parameters=OrderedDict(time="240", memory="32", cores="8"),
+    parameters_meta_info=OrderedDict(
+        cores=OrderedDict(description="number of available cores", data_type="int"),
+        memory=OrderedDict(description="memory in gigabytes", data_type="int"),
+        time=OrderedDict(description="time in minutes", data_type="int"),
+    ),
+)
+
+####################################################################################################
+# It is possible for flows to contain subflows. In this example, the auto-sklearn flow is a
+# subflow, this means that the subflow is entirely executed as part of this flow.
+# Using this modularity also allows your runs to specify which hyperparameters of the
```

Comment:
Is it also possible to define a custom flow without defining it as a subflow? So we need subflows when we want to do multiple runs with different hyperparameter settings for each subflow?

Reply:
It is possible to have custom flows with no subflows. Custom flows (such as this one) can be defined with their own hyperparameters (such as time or memory for this one).

Reply:
I understand, i am still a bit unsure of flow / custom flow / subflow differences. Maybe also additional value of using subflows instead of multiple custom flows, but that can be just me=)

Reply:
I think it is quite clear now.